internal/dataplane/kongstate/types.go

```diff
@@ -43,11 +43,29 @@ func (p *PortDef) CanonicalString() string {
 	return ImplicitPort
 }
 
+// ServiceBackendType is the type of the backend.
+type ServiceBackendType string
+
+const (
+	// ServiceBackendTypeKongServiceFacade means that the backend is an incubatorv1alpha1.KongServiceFacade.
+	ServiceBackendTypeKongServiceFacade ServiceBackendType = "KongServiceFacade"
```

Comment:
Why do not we define another constant to represent that the backend is a Kubernetes Service?

Reply:
My main point was to limit changes as much as possible, therefore making an empty value to be `Service` made sense to me. I initially had a separate const for the Service type, but if we'd like to fill all the struct initializations with the proper backend type, that would be more than 70 places to change in the code.

If you agree, I'm happy to create a separate issue so we can tackle that in a separate PR.

Reply:
The issue: https://github.com/Kong/kubernetes-ingress-controller/issues/5258